From ca6e434a7d2ee1110b70c0847c4cab1b22f94fb7 Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Tue, 24 Mar 2020 10:52:54 +0100
Subject: [PATCH 158/389] libfdisk: remove unwanted assert()

Signed-off-by: Karel Zak <kzak@redhat.com>
---
 libfdisk/src/label.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libfdisk/src/label.c b/libfdisk/src/label.c
index 0f6d85a..8e428da 100644
--- a/libfdisk/src/label.c
+++ b/libfdisk/src/label.c
@@ -614,7 +614,6 @@ void fdisk_label_set_changed(struct fdisk_label *lb, int changed)
  */
 int fdisk_label_is_changed(const struct fdisk_label *lb)
 {
-	assert(lb);
 	return lb ? lb->changed : 0;
 }
 
-- 
1.8.3.1

