From d29b7ced3b263ac9eda8a37703be0153ad67e882 Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Wed, 25 Jul 2018 12:02:28 +0200
Subject: [PATCH 312/686] lscpu: fix resource leak [coverity scan]

Signed-off-by: Karel Zak <kzak@redhat.com>
---
 sys-utils/lscpu.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sys-utils/lscpu.c b/sys-utils/lscpu.c
index 35df21c..7f5a9f9 100644
--- a/sys-utils/lscpu.c
+++ b/sys-utils/lscpu.c
@@ -795,8 +795,9 @@ read_hypervisor(struct lscpu_desc *desc, struct lscpu_modifier *mod)
 				else if ((features & XEN_FEATURES_PVH_MASK)
 								== XEN_FEATURES_PVH_MASK)
 					desc->virtype = VIRT_PARA;
-				fclose(fd);
 			}
+			if (fd)
+				fclose(fd);
 		}
 	} else if (read_hypervisor_powerpc(desc) > 0) {}
 
-- 
1.8.3.1

