From e6601a70ec4a56e99cb7361bb5d3a9c00072f9d8 Mon Sep 17 00:00:00 2001
From: Mark Sheppard <mark@ddf.net>
Date: Mon, 13 Aug 2018 16:45:01 +0100
Subject: [PATCH 342/686] lscpu: fixed part ID for ARM Cortex-M7

---
 sys-utils/lscpu-arm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys-utils/lscpu-arm.c b/sys-utils/lscpu-arm.c
index 22a9d6f..37b8f66 100644
--- a/sys-utils/lscpu-arm.c
+++ b/sys-utils/lscpu-arm.c
@@ -60,7 +60,7 @@ static const struct id_part arm_part[] = {
     { 0xc21, "Cortex-M1" },
     { 0xc23, "Cortex-M3" },
     { 0xc24, "Cortex-M4" },
-    { 0xc20, "Cortex-M7" },
+    { 0xc27, "Cortex-M7" },
     { 0xc60, "Cortex-M0+" },
     { 0xd01, "Cortex-A32" },
     { 0xd03, "Cortex-A53" },
-- 
1.8.3.1

